test/box-luatest/gh_4348_transactional_ddl_test.lua

```diff
@@ -0,0 +1,281 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group('gh-4348-transactional-ddl-test', {
+    {mvcc = false, engine = 'memtx'},
+    {mvcc = false, engine = 'vinyl'},
+    {mvcc = true, engine = 'memtx'},
+    {mvcc = true, engine = 'vinyl'},
```

Comment:
Nit: There's `t.helpers.matrix` helper that can simplify the test group definition a little bit:

```Lua
t.helpers.matrix({mvcc = {true, false}, engine = {'vinyl', 'memtx'}})
```

Anyway, I'm not sure that we actually need to run the tests for Vinyl - the code you patch is engine-independent.

Reply:
Yes, it seems like, made the code engine-independent.